src/ext/compatibility.h

```diff
@@ -0,0 +1,34 @@
+#ifndef DD_COMPATIBILITY_H
```

Comment:
❤️ I like moving this stuff to its own file and the file name make total sense. Nice work! We just gotta add this to the **package.xml** file. :)